app/models/order_balance.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class OrderBalance
+  def initialize(order)
+    @order = order
+  end
+
+  def label
+    to_f.negative? ? I18n.t(:credit_owed) : I18n.t(:balance_due)
+  end
+
+  def amount
+    Spree::Money.new(to_f, currency: order.currency)
+  end
+
+  def to_f
+    if customer_balance_enabled?
+      order.new_outstanding_balance
+    else
+      order.outstanding_balance
+    end
+  end
+
+  delegate :zero?, to: :to_f
+
+  private
+
+  attr_reader :order
+
+  def customer_balance_enabled?
+    OpenFoodNetwork::FeatureToggle.enabled?(:customer_balance, order.user)
```

Comment:
`order.user`? That would be nice but I thought that a feature toggle depends on the `current_user` which can be an admin looking at orders from others?

Reply:
yes, I forgot to explain this. In the context of all the affected views/mailers `current_user == order.user` or we can assume so. That's because these are actions that display a single order as opposed to an `:index`-like action.

That being said, if there's a specific context in which `current_user != order.user` we can always make it so that the user can be injected in the constructor and use that one instead. I haven't found it yet though but we'll see case by case.